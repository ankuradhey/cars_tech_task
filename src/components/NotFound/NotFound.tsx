import React, { FC } from "react";
import { Container } from "react-bootstrap";
import { useHistory } from "react-router-dom";
import Logo from "../../assets/logo.png";
import { Text } from "../Text";
import { Link } from "../Link";

export const NotFound: FC = () => {
    const history = useHistory();
    return (
        <Container
            className="d-flex flex-column justify-content-center align-items-center"
            style={{ minHeight: "calc(100vh - 160px)" }}
        >
            <img src={Logo} width="190px" alt="logo" className="margin-bottom-3 d-flex" />
            <Text size="heading" className="margin-bottom-3 d-flex">
                404 - Not Found
            </Text>
            <Text size="large" className="margin-bottom-3 d-flex">
                Sorry, the page you are looking for does not exist.
            </Text>
            <Text size="large" className="margin-bottom-2 d-flex">
                You can always go back to the &nbsp;
                <Link onClick={() => history.push("/")}> homepage</Link>
            </Text>
        </Container>
    );
};
