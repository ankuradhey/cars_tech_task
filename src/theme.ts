import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
