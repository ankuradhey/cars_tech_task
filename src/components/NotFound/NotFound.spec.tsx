import React from "react";
import { render } from "@testing-library/react";
import { NotFound } from "./NotFound";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";

describe("Button Component", () => {
    it("Should render Button component", () => {
        const { container, getByText } = render(
            <ThemeProvider theme={defaultTheme}>
                <NotFound />
            </ThemeProvider>
        );
        expect(getByText("404 - Not Found")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });
});
