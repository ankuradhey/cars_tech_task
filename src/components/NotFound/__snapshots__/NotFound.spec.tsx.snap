// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Should render Button component 1`] = `
<div
  class="d-flex flex-column justify-content-center align-items-center container"
  style="min-height: calc(100vh - 160px);"
>
  <img
    alt="logo"
    class="margin-bottom-3 d-flex"
    src="logo.png"
    width="190px"
  />
  <div
    class="sc-bdfBwQ fDCeuL margin-bottom-3 d-flex"
  >
    404 - Not Found
  </div>
  <div
    class="sc-bdfBwQ veibz margin-bottom-3 d-flex"
  >
    Sorry, the page you are looking for does not exist.
  </div>
  <div
    class="sc-bdfBwQ veibz margin-bottom-2 d-flex"
  >
    You can always go back to the Â 
    <button
      class="sc-gsTCUz ezYRfg"
    >
       homepage
    </button>
  </div>
</div>
`;
